### PR TITLE
[R2M] Add en_core_web_sm spacy model

### PR DESCRIPTION
**Pull request by sodre**
xref: #7728

### CODE REVIEW

**Comment by sodre** [comment]
@conda-forge/help-python 
This *noarch* recipe is ready to review. 

The Azure win_64 build is failing, but it works fine on AppVeyor. Either way, this should not matter because the final builds occur only on linux.

I would like to get feedback on the recipe since I need to create another 23 recipes that look almost exactly the same, with exception of the model size (small, medium, large) and language. 

cf. original issue request #7728

File: recipes/en_core_web_sm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{% set name = "en_core_web_sm" %}
+{% set version = "2.0.0" %}
+
+package:
+  name: {{ name|lower }}
```
Comment (mariusvniekerk) [inline recipes/en_core_web_sm/meta.yaml]: i'd probably recommend we have a more spacy specific prefix to this name here ?

**Comment by beckermr** [comment]
I had one thought here. Would it be worth using loops in jinja2 to combine all of these recipes into a single feedstock with multiple outputs? This might cut down on maintenance.

File: recipes/en_core_web_sm/meta.yaml (hunk 1 above)
Comment (sodre) [reply recipes/en_core_web_sm/meta.yaml]: how about spacy-model-en_core_web_sm?

**Comment by sodre** [comment]
> I had one thought here. Would it be worth using loops in jinja2 to combine all of these recipes into a single feedstock with multiple outputs? This might cut down on maintenance.

I thought about that originally, but ran into the issue where models can get updated at different times, e.g. the en_core_web_sm model is at version 2.0.0, while the "pt_<etc>" could be at version 1.0.0.

Would we still be able to handle that case in your approach?

**Comment by beckermr** [comment]
Yes, multiple outputs can have different versions. I don't see a problem if we have to bump the build number for everything either, but that is me.

https://conda.io/projects/conda-build/en/latest/source/define-metadata.html#outputs-section

File: recipes/en_core_web_sm/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+size:
+  - sm
+  - md
+  - lg
```
Comment (beckermr) [inline recipes/en_core_web_sm/conda_build_config.yaml]: You don't need multiple outputs and this build matrix. IDK the tradeoffs between the two, but we should only be using one mechanism I think.

**Comment by sodre** [comment]
@beckermr, @mariusvniekerk 

I think I fixed the two issues you both found:
  - I am using 'space-model' as a prefix for all packages.
  - I am using the conda build matrix to combine the small, medium and large models for a given language and type, in this case `en_core_web.`

I still don't know why the Azure win_64 build is failing while the AppVeyor build succeeds. If it is okay, I would like to ignore that failure for the moment since the package will only be built on Linux because of the `nonarch: python` tag.

File: recipes/en_core_web_sm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+{% set prefix = "spacy-model-" %}
+{% set name = "en_core_web" %}
+{% set version = "2.0.0" %}
+
+package:
+  name: {{ prefix }}{{ name|lower }}
+  version: {{ version }}
```
Comment (beckermr) [inline recipes/en_core_web_sm/meta.yaml]: Do you want all outputs to have the same version?

Comment (sodre) [reply recipes/en_core_web_sm/meta.yaml]: Yes, in this case all the english models have the same version. I will issue a separate PR for the other languages.

This should be ready to merge if you agree with it.

**Pull request merged**